# This ebuild was auto-generated by homeassistant script, and the script was not able to find a better candidate.
EAPI=8

DISTUTILS_USE_PEP517=setuptools
PYTHON_COMPAT=( python3_{12,13{,t}} )
GENERATED_IUSE="grpc"
IUSE="${GENERATED_IUSE}"

inherit distutils-r1 pypi

DESCRIPTION=""
HOMEPAGE="
  https://pypi.org/project/pinecone-client/"

LICENSE="Apache-2.0"
SLOT="0"
KEYWORDS="amd64 arm64"

REQUIRES_DIST="
	certifi>=2019.11.17
	googleapis-common-protos>=1.53.0; extra == "grpc"
	grpcio>=1.44.0; (python_version >= "3.8" and python_version < "3.11") and extra == "grpc"
	grpcio>=1.59.0; (python_version >= "3.11" and python_version < "4.0") and extra == "grpc"
	lz4>=3.1.3; extra == "grpc"
	pinecone-plugin-inference<2.0.0,>=1.0.3
	pinecone-plugin-interface<0.0.8,>=0.0.7
	protobuf<5.0,>=4.25; extra == "grpc"
	protoc-gen-openapiv2<0.0.2,>=0.0.1; extra == "grpc"
	tqdm>=4.64.1
	typing-extensions>=3.7.4
	urllib3>=1.26.0; python_version >= "3.8" and python_version < "3.12"
	urllib3>=1.26.5; python_version >= "3.12" and python_version < "4.0"
"
GENERATED_RDEPEND="${RDEPEND}
	>=dev-python/certifi-2019.11.17[${PYTHON_USEDEP}]
	grpc? ( >=dev-python/googleapis-common-protos-1.53.0[${PYTHON_USEDEP}] )
	grpc? ( >=dev-python/grpcio-1.59.0[${PYTHON_USEDEP}] )
	grpc? ( >=dev-python/lz4-3.1.3[${PYTHON_USEDEP}] )
	>=dev-python/pinecone-plugin-inference-1.0.3[${PYTHON_USEDEP}] <dev-python/pinecone-plugin-inference-2.0.0[${PYTHON_USEDEP}]
	>=dev-python/pinecone-plugin-interface-0.0.7[${PYTHON_USEDEP}] <dev-python/pinecone-plugin-interface-0.0.8[${PYTHON_USEDEP}]
	grpc? ( >=dev-python/protobuf-4.25[${PYTHON_USEDEP}] <dev-python/protobuf-5.0[${PYTHON_USEDEP}] )
	grpc? ( >=dev-python/protoc-gen-openapiv2-0.0.1[${PYTHON_USEDEP}] <dev-python/protoc-gen-openapiv2-0.0.2[${PYTHON_USEDEP}] )
	>=dev-python/tqdm-4.64.1[${PYTHON_USEDEP}]
	>=dev-python/typing-extensions-3.7.4[${PYTHON_USEDEP}]
	>=dev-python/urllib3-1.26.5[${PYTHON_USEDEP}]
"
RDEPEND="${GENERATED_RDEPEND}"

distutils_enable_tests pytest
